src/oemof/network/energy_system.py

```diff
@@ -168,17 +168,21 @@ def groups(self):
             (
                 g(n, gs)
                 for g in self._groupings
-                for n in self.nodes[self._first_ungrouped_node_index_ :]
+                for n in list(self.nodes)[self._first_ungrouped_node_index_ :]
             ),
             maxlen=0,
         )
         self._first_ungrouped_node_index_ = len(self.nodes)
         return self._groups
 
     @property
-    def nodes(self):
+    def node(self):
```

Comment:
Maybe confusing, as I would not expect an iterable (from single name).
Maybe, instead rewrite as `def node(self, key)` ?

Reply:
As I can see that you want to iterate over labels aswell, maybe implement a function `node_labels()` (or similar) as well?

Reply:
The idea is that you can do
```python
node_named_name = model.node["name"]
```

Reply:
for me it's okay this way